.vscode/settings.json

```diff
@@ -4,6 +4,9 @@
   // When opening a file, `editor.tabSize` and `editor.insertSpaces` will NOT be detected based on the file contents.
   "editor.detectIndentation": false,
   "editor.formatOnSave": true,
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
+  },
```

Comment:
I've had problems in the past with this. It was somehow clashing with prettier and I couldn't debug why. in the end I resolved it by using this:
```json
"editor.codeActionsOnSave": {
  "source.organizeImports": true
},
```
This was relevant only for sorting, for the rest I am using `"editor.formatOnSave": true`

Reply:
problem with `source.organizeImports` is that VScode uses its own custom logic which is not consistent with lint setup. we can disable this if such an issue occurs.

`editor.formatOnSave` is already enabled for prettier run

Reply:
I am perfectly okay with that. Just wanted to mention it because I've had problems with it in the past.